pyiceberg/table/metadata.py

```diff
@@ -178,6 +178,12 @@ class TableMetadataCommonFields(IcebergBaseModel):
     to be used for arbitrary metadata. For example, commit.retry.num-retries
     is used to control the number of commit retries."""
 
+    @field_validator("properties", mode='before')
+    @classmethod
+    def transform_dict_value_to_str(cls, dict: Dict[str, Any]) -> Dict[str, str]:
+        assert None not in dict.values(), "None type is not a supported value in properties"
```

Comment:
We try to avoid asserts outside of `tests/`. Could you raise a `ValueError` instead?